helpers/setup_json_generator.py

```diff
@@ -0,0 +1,297 @@
+
+    def add_storagerouter(self, storagerouter_ip, hostname):
+        try:
+            socket.inet_aton(storagerouter_ip)
+        except TypeError:
+            raise ValueError('Invalid ip adress provided: {0}'.format(storagerouter_ip))
+        if not isinstance(hostname,str):
+            raise ValueError('Invalid hostname provided: {0} must be a string, not {1}'.format(hostname,type(hostname)))
+        sr_dict = {}
+        sr_dict[storagerouter_ip] = {"hostname": hostname,
+                                     "domains": {},
+                                     "disks": {},
+                                     "vpools": {}
+                                     }
+        self.json['setup']['storagerouters'].update(sr_dict)
+
+    def _add_disk_to_sr(self, storagerouter_ip, name, roles):
+        if not storagerouter_ip in self.json['setup']['storagerouters'].keys():
+            raise ValueError('Storagerouter with IP {0} not found in json'.format(storagerouter_ip))
+        d_dict = {}
+        d_dict[name] = {"roles": roles}
+        self.json['setup']['storagerouters'][storagerouter_ip]['disks'].update(d_dict)
+
+    def _add_domain_to_sr(self, storagerouter_ip, name, recovery=False):
+        if not storagerouter_ip in self.json['setup']['storagerouters'].keys():
+            raise ValueError('Storagerouter with ip {0} not found in json'.format(storagerouter_ip))
+        if not name in self._domains:
+            raise ValueError('Invalid domain passed: {}'.format(name))
+        path = self.json['setup']['storagerouters'][storagerouter_ip]['domains']
+
+        if recovery is False:
+            if 'domain_guids' not in path.keys():
+                path['domain_guids'] = [name]
+            else:
+                path['domain_guids'].append(name)
+        else:
+            if 'recovery_domain_guids' not in path.keys():
+                path['recovery_domain_guids'] = [name]
+            else:
+                path['recovery_domain_guids'].append(name)
+
+    def add_domain(self, domain):
+        if not isinstance(domain, str):
+            raise ValueError("domain is no string")
+        self._domains.append(domain)
+        self.json['setup']['domains'].append(domain)
+
+    def add_backend(self, name, domains=None, scaling='LOCAL'):
+        if domains is None:
+            domains = []
+        for domain_name in domains:
+            if domain_name not in self._domains:
+                raise ValueError('Invalid domain passed: {}'.format(domain_name))
+
+        be_dict = {'name': name,
+                   'domains': {'domain_guids': domains},
+                   'scaling': scaling,
+                   'presets': [],
+                   'osds': {}
+                   }
+        self._backends.append(be_dict['name'])
+        self.json['setup']['backends'].append(be_dict)
+
+    def _add_preset_to_backend(self, backend_name, preset_name, policies, compression='snappy', encryption=None, fragment_size=2097152):
```

Comment:
Public function?
Add validation for policies, compression, encryption, fragment size and preset_name